pkg/db/postgres_container_runtime.go

```diff
@@ -0,0 +1,87 @@
+package db
+
+import (
+	"context"
+	"github.com/docker/go-connections/nat"
+	"github.com/kyma-incubator/reconciler/pkg/test"
+	"github.com/spf13/viper"
+	"strconv"
+)
+
+type PostgresContainerRuntime struct {
+	debug bool
+
+	env    postgresEnvironment
+	encKey string
+
+	ContainerBootstrap
+	ConnectionFactory
+}
+
+type Migrations string
+
+var NoMigrations Migrations = ""
+
+func RunPostgresContainer(ctx context.Context, migrations Migrations, debug bool) (*PostgresContainerRuntime, error) {
+	configFile, err := test.GetConfigFile()
+
+	if err != nil {
+		return nil, err
+	}
+
+	viper.SetConfigFile(configFile)
+
+	configReadError := viper.ReadInConfig()
+
+	if configReadError != nil {
+		return nil, err
+	}
+
+	encKey, encryptError := readEncryptionKey()
+
+	if encryptError != nil {
+		return nil, err
+	}
+
+	env := getPostgresEnvironment()
+
+	cont, bootstrapError := BootstrapNewPostgresContainer(ctx, env)
+
+	if bootstrapError != nil {
+		return nil, err
+	}
+
+	externalPort, portFetchError := cont.MappedPort(ctx, nat.Port(strconv.Itoa(env.port)))
+
+	if portFetchError != nil {
+		panic(portFetchError)
+	}
+
+	connectionFactory := postgresConnectionFactory{
+		host:          env.host,
+		port:          externalPort.Int(),
+		database:      env.database,
+		user:          env.user,
+		password:      env.password,
+		sslMode:       env.sslMode,
+		encryptionKey: encKey,
+		migrationsDir: string(migrations),
+		blockQueries:  true,
+		logQueries:    true,
+		debug:         debug,
+	}
+
+	shouldMigrate := len(string(migrations)) > 0
```

Comment:
Why this `migrations` define as string but not boolean?

Reply:
This allows each container runtime to have a dedicated migrations directory that can be targeted, allowing us to define separate container instances with different schema migrations (e.g. for A/B testing or for compatibility checks): https://github.com/kyma-incubator/reconciler/pull/886/files#diff-927e18379e6faf9e08bc0bf2d05805183bfb4e12ac3f566cb6d774179bede063R34-R37

This is then used in https://github.com/kyma-incubator/reconciler/pull/886/files#diff-5ad72269b02f5a6b48538fc500a3b237d5a3be729838dc51affb780a7c097270R68 to overwrite the value from the environment settings.

Reply:
OK, I haven't read to the real usage yet. Then this totally make sense, just one suggestion, maybe renaming the type to `MigrationPath` or `MigrationDirectory` to have better understanding?